source/SpinalHDL/Data types/AFix.rst

```diff
@@ -17,50 +17,52 @@ AFix can be created using bit sizes or exponents:
 
 .. code-block:: scala
 
-  U(12 bits)        // U12.0
-  UQ(8 bits, 4 bits) // U8.4
-  U(8 exp, 12 bits) // U8.4
-  U(8 exp, -4 exp) // U8.4
-  U(8 exp, 4 exp)  // U8.-4
+  AFix.U(12 bits)        // U12.0
+  AFix.UQ(8 bits, 4 bits) // U8.4
+  AFix.U(8 exp, 12 bits) // U8.4
+  AFix.U(8 exp, -4 exp) // U8.4
+  AFix.U(8 exp, 4 exp)  // U8.-4
 
-  S(12 bits)        // S11 + sign
-  SQ(8 bits, 4 bits) // S8.4 + sign
-  S(8 exp, 12 bits) // S8.3 + sign
-  S(8 exp, -4 exp) // S8.4 + sign
+  AFix.S(12 bits)        // S11 + sign
+  AFix.SQ(8 bits, 4 bits) // S8.4 + sign
+  AFix.S(8 exp, 12 bits) // S8.3 + sign
+  AFix.S(8 exp, -4 exp) // S8.4 + sign
 
 
 These will have representable ranges for all bits.
 
 For example:
 
-``U(12 bits)`` will have a range of 0 to 4095.
+``AFix.U(12 bits)`` will have a range of 0 to 4095.
 
-``SQ(8 bits, 4 bits)`` will have a range of -4096 (-256) to 4095 (255.9375)
+``AFix.SQ(8 bits, 4 bits)`` will have a range of -4096 (-256) to 4095 (255.9375)
 
-``U(8 exp, 4 exp)`` will have a range of 0 to 256
+``AFix.U(8 exp, 4 exp)`` will have a range of 0 to 256
 
 
 Custom range ``AFix`` values can be created be directly instantiating the class.
 
 .. code-block:: scala
 
-  AFix(0, 4096, 0 exp)
-  AFix(-256, 256, -2 exp)
-  AFix(8, 16, 2 exp)
+  class AFix(val maxValue: BigInt, val minValue: BigInt, val exp: ExpNumber)
+
+  AFix.AFix(4096, 0, 0 exp)     // [0 to 4096, 2^0]
+  AFix.AFix(256, -256, -2 exp)  // [-256 to 256, 2^-2]
+  AFix.AFix(16, 8, 2 exp)       // [8 to 16, 2^2]
 
 
 The ``maxValue`` and ``minValue`` stores what backing integer values are representable.
 These values represent the true fixed-point value after multiplying by ``2^exp``.
 
-``U(2 exp, -1 exp)`` can represent:
+``AFix.U(2 exp, -1 exp)`` can represent:
```

Comment:
isn't it 0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5 ?

Reply:
IS this resolved.. I don't see any comment/or modification on it. Should i start the merge? (if yes just click resolve conversation)